dashboard/modules/job/job_head.py

```diff
@@ -56,9 +71,45 @@ async def _parse_and_validate_request(
                 status=aiohttp.web.HTTPBadRequest.status_code,
             )
 
-    def job_exists(self, job_id: str) -> bool:
-        status = self._job_manager.get_job_status(job_id)
-        return status is not None
+    async def find_job_by_ids(self, job_or_submission_id: str) -> Optional[JobDetails]:
+        """
+        Attempts to find the job with a given submission_id or job id.
+        """
+        # First try to find by job_id
+        driver_jobs, submission_job_drivers = await self._get_driver_jobs()
+        job = driver_jobs.get(job_or_submission_id)
+        if job:
+            return job
+        # Try to find a driver with the given id
+        submission_id = next(
+            (
+                id
+                for id, driver in submission_job_drivers.items()
+                if driver.id == job_or_submission_id
+            ),
+            None,
+        )
+
+        if not submission_id:
+            # If we didn't find a driver with the given id,
+            # then lets try to search for a submission with given id
+            submission_id = job_or_submission_id
+
+        job_info = await asyncio.get_running_loop().run_in_executor(
+            self._executor, lambda: self._job_manager.get_job_info(submission_id)
```

Comment:
Can you double check if `get_job_info ` is thread-safe?

Reply:
@rkooo567 this is essentially directly an internal_kv call, so it should be thread safe.

In general, I'd much prefer to refactor the job manager to make it `asyncio` natively instead of using `run_in_executor` though because it's easy to introduce unexpected behaviors with this. @alanwguo is there anything blocking this? I would like to do it as an immediate follow-up if not in this PR.

Reply:
I'll do it as a follow-up. No blockers, just want to avoid refactoring a bunch of code in this PR.